src/browser/history.js

```diff
@@ -0,0 +1,321 @@
+    [PouchDB.revision]: Maybe(Revision),
+
+    type: Type('Quote'),
+    uri: URI,
+    content: String,
+    tags: List(TagName)
+  });
+  Quote.storeID = 'quotes';
+
+  Quote.construct = async(function* ({uri, content}) {
+    const hash = yield sha(content);
+    return Quote({[PouchDB.id]: `quote/${uri}/${hash}`, uri, content});
+  });
+
+
+  /**
+  {
+    _id: "tag/haskell",
+    description: "Haskell programming language",
+    name: "haskell",
+    items: [
+      "quote/W29iamVjdCBBcnJheUJ1ZmZlcl0="
+      "page/http://learnyouahaskell.com"
+    ]
+  }
+  **/
+
+  const Tag = Record({
+    [PouchDB.id]: ID,
+    [PouchDB._rev]: Maybe(Revision),
+    type: Type("Tag"),
+    name: TagName,
+    items: List(ID)
+  });
+  Tag.storeID = 'tags';
+
+  Tag.add = (tag, item) =>
+    tag.update('items', include(item[PouchDB.id]));
+
+  Tag.remove = (tag, item) =>
+    tag.update('items', exclude(item[PouchDB.id]));
+
+  Tag.tag = (item, tagName) => item.update('tags', include(tagName));
+
+  Tag.untag = (item, tagName) => item.update('tags', exclude(tagName));
+
+  const Top = Record({
+    [PouchDB.id]: ID('top/pages'),
+    [PouchDB.revision]: Maybe(Revision),
+    pages: List(Page)
+  });
+  Top.storeID = Page.storeID;
+  Top.sample = (page, limit) => top => top.update('pages', pages => {
+    const index = pages.findIndex(x => x[PouchDB.id] === page[PouchDB.id])
+    return pages.set(index < 0 ? pages.size : index, page)
+                .sortBy(Page.frequency)
+                .take(limit);
+  });
+
+  const clear = async(function*({stores}) {
+    yield stores.pages.destroy();
+    yield stores.quotes.destroy();
+    yield stores.tags.destroy();
+  });
+
+  let taskID = 0;
+
+  const scheduleEdit = async(function*(history, record, transform) {
+    try {
+      // wait for last scehduled edit to complete.
+      yield history.editQueue[record[PouchDB.id]];
+    } catch (error) {
+      // If last operation faild log an error, but proceed to a next
+      // operation.
+      console.error(error);
+    } finally {
+      const store = history.stores[record.constructor.storeID];
+      return edit(record, transform, store);
+    }
+  })
+
+  // History
+  class History {
+    static defaults() {
+      return {
+        pagesStoreName: "pages",
+        quotesStoreName: "quotes",
+        tagsStoreName: "tags",
+        topPageLimit: 6,
+        trackTopPages: false
+      }
+    }
+    constructor(options={}) {
+      this.onPageChange = this.onPageChange.bind(this);
+      this.onTopPagesChange = this.onTopPagesChange.bind(this);
+
+      this.options = Object.assign(History.defaults(), options);
+      const {pagesStore, quotesStore, tagsStore,
+             pagesStoreName, quotesStoreName, tagsStoreName} = this.options;
+
+      this.stores = {
+        pages: pagesStore || new PouchDB(pagesStoreName),
+        quotes: quotesStore || new PouchDB(quotesStoreName),
+        tags: tagsStore || new PouchDB(tagsStoreName)
```

Comment:
This may be a work around for lack of PouchDB views? Just thought I would call out that Pouch is designed to be used in such a way that you dump everything in one database and create view indexes for your different "buckets".